components/ai_chat/resources/page/components/model_selector/index.tsx

```diff
@@ -134,6 +135,18 @@ export function ModelSelector() {
         </div>
       </Button>
 
+      {aiChatContext.isAIChatAgentProfile
+        && aiChatContext.isAIChatAgentProfileFeatureEnabled && (
```

Comment:
I would rather base this on `conversationCapability` so that we won't have to come back here and make it dynamic after we allow the user to toggle this capability even within the agent profile, but I'll settle for a TODO. The TODO can point to https://github.com/brave/brave-browser/issues/49261